File: services/auth/go/cmd/cors.go
package cmd

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

func cors() gin.HandlerFunc {
	f := func(c *gin.Context, origin string) {
		c.Header("Access-Control-Allow-Origin", origin)
		c.Header("Access-Control-Allow-Methods", "POST, GET")
		headers := c.Request.Header.Get("Access-Control-Request-Headers")
		c.Header("Access-Control-Allow-Headers", headers)
		c.Header("Access-Control-Allow-Credentials", "true")
		c.Header("Access-Control-Max-Age", "86400")
		c.Writer.Header().Add("Vary", "Origin, Access-Control-Request-Method")
	}

	return func(c *gin.Context) {
		origin := c.Request.Header.Get("Origin")
		if c.Request.Method == http.MethodOptions {
			f(c, origin)

			c.Header("Content-Length", "0")
			c.AbortWithStatus(http.StatusNoContent)
		}

		if origin != "" {
			f(c, origin)
		}

		c.Next()
	}
}

File: examples/demos/backend/functions/cors.ts
import type { Request, Response } from "express";

export default (req: Request, res: Response) => {
  res.setHeader("Access-Control-Allow-Origin", "*");
  res.setHeader(
    "Access-Control-Allow-Methods",
    "GET, POST, PUT, DELETE, OPTIONS",
  );
  res.setHeader(
    "Access-Control-Allow-Headers",
    "Content-Type, Authorization",
  );

  if (req.method === "OPTIONS") {
    return res.status(204).end();
  }

  res.status(200).json({
    message: "Hello from a CORS-enabled function!",
    origin: req.headers.origin || "no origin header",
    method: req.method,
  });
};
